samcli/local/apigw/local_apigw_service.py

```diff
@@ -526,6 +527,20 @@ def _parse_v2_payload_format_lambda_output(lambda_output: str, binary_types, fla
         status_code = json_output.get("statusCode") or 200
         headers = Headers(json_output.get("headers") or {})
 
+        # cookies is a new field in payload format version 2.0 (a list)
+        # https://docs.aws.amazon.com/apigateway/latest/developerguide/http-api-develop-integrations-lambda.html
+        # we need to move cookies to Set-Cookie headers.
+        if isinstance(json_output.get("cookies"), list):
+            # add all cookies to Set-Cookie header
+            if "Set-Cookie" not in headers:
+                headers["Set-Cookie"] = ""
+            for cookie in json_output.get("cookies"):
+                headers["Set-Cookie"] = headers["Set-Cookie"] + ";" + cookie
+            # remove the first semi colon and trim the header
+            if headers["Set-Cookie"][0] == ";":
+                headers["Set-Cookie"] = headers["Set-Cookie"][1:]
+            headers["Set-Cookie"] = headers["Set-Cookie"].strip()
```

Comment:
nit: There are several branches that we can use "join" to simplify:

```suggestion
            headers["Set-Cookie"] = (
                ";".join([
                    cookie for cookie in [headers.get("Set-Cookie")] + json_output.get("cookies") if cookie
                ])
            ).strip()
```